src/library_websocket.js

```diff
@@ -0,0 +1,403 @@
+    return {{{ cDefine('EMSCRIPTEN_RESULT_SUCCESS') }}};
+  },
+
+  emscripten_websocket_get_extensions__proxy: 'sync',
+  emscripten_websocket_get_extensions__sig: 'iiii',
+  emscripten_websocket_get_extensions: function(socketId, extensions, extensionsLength) {
+    var socket = WS.sockets[socketId];
+    if (!socket) {
+#if WEBSOCKET_DEBUG
+      console.error('emscripten_websocket_get_extensions(): Invalid socket ID ' + socketId + ' specified!');
+#endif
+      return {{{ cDefine('EMSCRIPTEN_RESULT_INVALID_TARGET') }}};
+    }
+    return {{{ cDefine('EMSCRIPTEN_RESULT_SUCCESS') }}};
+  },
+
+  emscripten_websocket_get_extensions_length__proxy: 'sync',
+  emscripten_websocket_get_extensions_length__sig: 'iii',
+  emscripten_websocket_get_extensions_length: function(socketId, extensionsLength) {
+    var socket = WS.sockets[socketId];
+    if (!socket) {
+#if WEBSOCKET_DEBUG
+      console.error('emscripten_websocket_get_extensions_length(): Invalid socket ID ' + socketId + ' specified!');
+#endif
+      return {{{ cDefine('EMSCRIPTEN_RESULT_INVALID_TARGET') }}};
+    }
+    return {{{ cDefine('EMSCRIPTEN_RESULT_SUCCESS') }}};
+  },
+
+  emscripten_websocket_get_protocol__proxy: 'sync',
+  emscripten_websocket_get_protocol__sig: 'iiii',
+  emscripten_websocket_get_protocol: function(socketId, protocol, protocolLength) {
+    var socket = WS.sockets[socketId];
+    if (!socket) {
+#if WEBSOCKET_DEBUG
+      console.error('emscripten_websocket_get_protocol(): Invalid socket ID ' + socketId + ' specified!');
+#endif
+      return {{{ cDefine('EMSCRIPTEN_RESULT_INVALID_TARGET') }}};
+    }
+    return {{{ cDefine('EMSCRIPTEN_RESULT_SUCCESS') }}};
+  },
+
+  emscripten_websocket_get_protocol_length__proxy: 'sync',
+  emscripten_websocket_get_protocol_length__sig: 'iii',
+  emscripten_websocket_get_protocol_length: function(socketId, protocolLength) {
+    var socket = WS.sockets[socketId];
+    if (!socket) {
+#if WEBSOCKET_DEBUG
+      console.error('emscripten_websocket_get_protocol_length(): Invalid socket ID ' + socketId + ' specified!');
+#endif
+      return {{{ cDefine('EMSCRIPTEN_RESULT_INVALID_TARGET') }}};
+    }
+    return {{{ cDefine('EMSCRIPTEN_RESULT_SUCCESS') }}};
+  },
+
+  emscripten_websocket_get_url__proxy: 'sync',
+  emscripten_websocket_get_url__sig: 'iiii',
+  emscripten_websocket_get_url: function(socketId, url, urlLength) {
+    var socket = WS.sockets[socketId];
+    if (!socket) {
+#if WEBSOCKET_DEBUG
+      console.error('emscripten_websocket_get_url(): Invalid socket ID ' + socketId + ' specified!');
+#endif
+      return {{{ cDefine('EMSCRIPTEN_RESULT_INVALID_TARGET') }}};
+    }
+    return {{{ cDefine('EMSCRIPTEN_RESULT_SUCCESS') }}};
+  },
+
+  emscripten_websocket_get_url_length__proxy: 'sync',
+  emscripten_websocket_get_url_length__sig: 'iii',
+  emscripten_websocket_get_url_length: function(socketId, urlLength) {
+    var socket = WS.sockets[socketId];
```

Comment:
That would regress on release mode code size, because the `if (!socket) return {{{ cDefine('EMSCRIPTEN_RESULT_INVALID_TARGET') }}};` part would still need to be in the caller side. So while it would reduce duplication of the `#if WEBSOCKET_DEBUG` part, it would make the code larger  by creating a redundant function. (`var socket = WS.sockets[socketId];` minifies probably to exact same length as `var socket = WS.get(socketId);`, but the function approach would have an extra `WS: { get: function(socketId) { return this.sockets[socketId]; } }` part to it.

Ideal solution to avoid code duplication here would be to use macros:

```
#if WEBSOCKET_DEBUG
#define WS_GET(socketId, name) WS.sockets[socketId]; \
    if (!socket) { \
      console.error(name + '(): Invalid socket ID ' + socketId + ' specified!'); \
      return {{{ cDefine('EMSCRIPTEN_RESULT_INVALID_TARGET') }}}; \
    }
#else
#define WS_GET(socketId, name) WS.sockets[socketId]; \
    if (!socket) return {{{ cDefine('EMSCRIPTEN_RESULT_INVALID_TARGET') }}};
#endif

and then in all parts, use `var socket = WS_GET(socketId);`. (But that would require support for full C preprocessing)